py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/batcher.py

```diff
@@ -34,7 +37,25 @@ def __str__(self) -> str:
 
     def add_file(self, filename: str, filesize: int) -> None:
         """Add a file to this batch"""
-        self.filenames.append(filename)
+        if (
+            self.prefix != ""
+            and filename.startswith(self.prefix)
+            and filename.endswith(self.suffix)
+        ):
+            filename = filename.removeprefix(self.prefix)
+            filename = filename.removesuffix(self.suffix)
+            self.filenames.append(filename)
+
```

Comment:
in the regular case, we can just lop the prefix and suffix off of our file each time and add the unique part to filenames.